### PR TITLE
Add javadoc for Show Catalogs Flow

### PR DESCRIPTION
**Pull request by anthonylouisbsb**
# Pull Request Merge Checklist
           
Issue identifier related to this PR: [700](https://github.com/s1mbi0se/simbiose_knowledge_base/issues/700)

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/execution/SqlTask.java [hunk 1]
```diff
@@ -495,6 +495,18 @@ public boolean isFinished()
...
+         * Gets the metadata about the executed task.
+         * <p>
+         * The metadata contains:
+         * - The {@link io.prestosql.operator.Driver} set that compounds
+         * the task.
+         * - The buffer that stores the output result.
+         * - The metadata about the {@link io.prestosql.metadata.Split} where
+         * query is executed.
+         *
+         * @return the metadata about executed task.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/SqlTask.java]:
```suggestion
         * @return the metadata about the executed task.
```

File: presto-main/src/main/java/io/prestosql/execution/TaskInfo.java [hunk 2]
```diff
@@ -61,6 +61,17 @@ public TaskInfo(@JsonProperty("taskStatus") TaskStatus taskStatus,
         this.needsPlan = needsPlan;
     }
 
+    /**
+     * Gets the information about a task execution.
+     * <p>
+     * Some returned information:
+     * - Task state(running, aborted, planned, etc.)
+     * - Task identifier
+     * - Node where task is being executed
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/TaskInfo.java]:
```suggestion
     * - Node where the task is being executed
```

File: presto-main/src/main/java/io/prestosql/execution/TaskInfo.java [hunk 3]
```diff
@@ -61,6 +61,17 @@ public TaskInfo(@JsonProperty("taskStatus") TaskStatus taskStatus,
...
+     * - Task's used memory
+     *
+     * @return the information about task execution
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/TaskInfo.java]:
```suggestion
     * @return the information about the task execution
```

File: presto-main/src/main/java/io/prestosql/execution/buffer/OutputBufferMemoryManager.java [hunk 4]
```diff
@@ -108,6 +108,12 @@ public synchronized void updateMemoryUsage(long bytesAdded)
         return bufferBlockedFuture;
     }
 
+    /**
+     * Allows task execution even if the buffer is full.
+     * <p>
+     * It is used when some query fails because the buffer is full, to complete
+     * the corresponding future.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/buffer/OutputBufferMemoryManager.java]: Can we explain better this description?

I couldn't understand exactly what the method is doing and what is its purpose, for example:
- Why this output buffer can't be full?
- What is this flag "blockOnFull" being set?

I believe that this information is important so I can understand why this method exists. Can we improve it?

File: presto-main/src/main/java/io/prestosql/execution/buffer/OutputBufferMemoryManager.java [hunk 5]
```diff
@@ -137,6 +143,15 @@ private synchronized boolean isBufferFull()
         return bufferedBytes.get() > maxBufferedBytes && blockOnFull.get();
     }
 
+    /**
+     * Indicates if system could reserve memory to execute a task.
+     * <p>
+     * The {@code blockedOnMemory} future should not be cancelled, or
+     * the server will allocate memory even if memory pools are full.
+     *
+     * @return a flag which indicates that system could allocate necessary
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/buffer/OutputBufferMemoryManager.java]:
```suggestion
     * @return a flag which indicates if the system can allocate necessary
```

File: presto-main/src/main/java/io/prestosql/execution/buffer/OutputBufferMemoryManager.java [hunk 6]
```diff
@@ -137,6 +143,15 @@ private synchronized boolean isBufferFull()
         return bufferedBytes.get() > maxBufferedBytes && blockOnFull.get();
     }
 
+    /**
+     * Indicates if system could reserve memory to execute a task.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/buffer/OutputBufferMemoryManager.java]:
```suggestion
     * Indicates if the system can reserve memory to execute a task.
```

File: presto-main/src/main/java/io/prestosql/server/TaskResource.java [hunk 7]
```diff
@@ -182,6 +182,16 @@ public void getTaskInfo(
                 .withTimeout(timeout);
     }
 
+    /**
+     * Receives GET HTTP requests and returns the metadata about a task.
+     *
+     * @param taskId the task's identifier
+     * @param currentState the task current state
+     * @param maxWait the time that server must wait until
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/TaskResource.java]:
```suggestion
     * @param maxWait the time that the server should wait until
```

File: presto-main/src/main/java/io/prestosql/sql/planner/PartitioningScheme.java [hunk 8]
```diff
@@ -91,6 +91,11 @@ public Partitioning getPartitioning()
         return outputLayout;
     }
 
+    /**
+     * Gets a column that is used in a task.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/PartitioningScheme.java]: Makes sense to make it clear that the column hash is being returned?
```suggestion
     * Gets the hash of the column used in a task.
```

File: presto-main/src/main/java/io/prestosql/sql/planner/PartitioningScheme.java [hunk 9]
```diff
@@ -91,6 +91,11 @@ public Partitioning getPartitioning()
...
+     *
+     * @return a used column in a task
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/PartitioningScheme.java]: the hash of the column used in a task.

File: presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java [hunk 10]
```diff
@@ -66,6 +66,18 @@ public TypeProvider getTypes()
         return Optional.ofNullable(nodeInfo.get(id));
     }
 
+    /**
+     * Adds a node in query plan tree.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java]:
```suggestion
     * Adds a node in the query plan tree.
```

File: presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java [hunk 11]
```diff
@@ -66,6 +66,18 @@ public TypeProvider getTypes()
...
+     * <p>
+     * A query plan tree represents the data that must be retrieved by a query,
+     * with each node representing an operation. As an example of sql plan tree:
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java]:
```suggestion
     * with each node representing an operation. As an example of a sql plan tree:
```

File: presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java [hunk 12]
```diff
@@ -66,6 +66,18 @@ public TypeProvider getTypes()
...
+     * - Aggregate(FINAL)[regionkey] => [regionkey:bigint, count:bigint]
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java:74]: Makes sense to add pipes like this so it looks more  like a tree?
```suggestion
     * |- Aggregate(FINAL)[regionkey] => [regionkey:bigint, count:bigint]
```

File: presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java [hunk 13]
```diff
@@ -66,6 +66,18 @@ public TypeProvider getTypes()
...
+     * -- LocalExchange[HASH][$hashvalue] ("regionkey") => regionkey:bigint, count_8:bigint, $hashvalue:bigint
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java]: Makes sense to add pipes like this so it looks more like a tree?
```suggestion
     *   |-- LocalExchange[HASH][$hashvalue] ("regionkey") => regionkey:bigint, count_8:bigint, $hashvalue:bigint
```

File: presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java [hunk 14]
```diff
@@ -66,6 +66,18 @@ public TypeProvider getTypes()
...
+     * -----RemoteExchange[REPARTITION][$hashvalue_9] => regionkey:bigint, count_8:bigint, $hashvalue_9:bigint
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java]: Makes sense to add pipes like this so it looks more like a tree?
```suggestion
     *      |--RemoteExchange[REPARTITION][$hashvalue_9] => regionkey:bigint, count_8:bigint, $hashvalue_9:bigint
```

File: presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java [hunk 15]
```diff
@@ -66,6 +66,18 @@ public TypeProvider getTypes()
...
+     * --------Project[] => [regionkey:bigint, count_8:bigint, $hashvalue_10:bigint]
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java]: Makes sense to add pipes like this so it looks more like a tree?
```suggestion
     *        |-----Project[] => [regionkey:bigint, count_8:bigint, $hashvalue_10:bigint]
```

File: presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java [hunk 16]
```diff
@@ -66,6 +66,18 @@ public TypeProvider getTypes()
...
+     *
+     * @param node an object with query a query plan tree node's metadata
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanRepresentation.java]:
```suggestion
     * @param node an object with a query plan tree node's metadata
```

**Review by jpedroantunes — CHANGES_REQUESTED** [review]

File: presto-main/src/main/java/io/prestosql/sql/planner/PartitioningScheme.java (hunk 8 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/sql/planner/PartitioningScheme.java]: done

File: presto-main/src/main/java/io/prestosql/execution/buffer/OutputBufferMemoryManager.java (hunk 4 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/execution/buffer/OutputBufferMemoryManager.java]: changed to 
```
/**
     * Allows task execution even if the buffer is full.
     * <p>
     * By default each task has an buffer to hold the output data,
     * if this buffer reaches the maximum quantity of bytes, the task is blocked.
     * This method allows a task to finish its execution even if the buffer
     * size limit is reached.
     */
```

**Review by jpedroantunes — APPROVED** [review]
